server/src/main/java/org/opensearch/common/util/ThrowableTranslateUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.opensearch.common.CheckedRunnable;
+import org.opensearch.common.CheckedSupplier;
+import org.opensearch.index.store.remote.fc.FileCache;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+/**
+ * Utils to translate/cast throwable instances
+ *
+ * @opensearch.internal
+ */
+final public class ThrowableTranslateUtils {
+    private static final Logger logger = LogManager.getLogger(FileCache.class);
+
+    /**
+     * Run passed runnable and catch exception and translate exception into runtime exception using
+     * {@link ThrowableTranslateUtils#asRuntimeException(Throwable)}
+     * @param supplier to run
+     */
+    public static <R, E extends Exception> R catchAsRuntimeException(CheckedSupplier<R, E> supplier) {
+        try {
+            return supplier.get();
+        } catch (Exception e) {
+            logger.error(() -> new ParameterizedMessage("translating exception into runtime exception"), e);
+            throw asRuntimeException(e);
```

Comment:
Log-and-throw is generally an anti-pattern because whoever ends up catching this usually does something sensible with it (often logging it, resulting in double-logging). I really don't like codifying this behavior in a public utility.

Reply:
yep I agree this should not be a public utility will try to eliminate this